test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -142,16 +144,17 @@ public class HubEndpointTests
             }
         }
 
-        [Fact]
-        public async Task HubMethodsAreCaseInsensitive()
+        [Theory]
+        [MemberData(nameof(HubTypes))]
+        public async Task HubMethodsAreCaseInsensitive(Type hubType)
         {
             var serviceProvider = CreateServiceProvider();
 
-            var endPoint = serviceProvider.GetService<HubEndPoint<MethodHub>>();
+            dynamic endPoint = serviceProvider.GetService(GetEndPointType(hubType));
```

Comment:
I'm not a fan of this. I get why, but I'd rather fix it the right way in this PR. If we need an interface for `HubEndPoint`, maybe we should just do that. Our use of `dynamic` should be limited only to instances of the proxies in `DynamicHub`.